src/inputs/TextField/index.tsx

```diff
@@ -14,11 +14,28 @@ type Props = {
   startAdornment?: React.ReactElement;
   endAdornment?: React.ReactElement;
   className?: string;
-} & React.InputHTMLAttributes<HTMLInputElement>;
+  color?: 'primary' | 'secondary';
+};
 
-const CustomTextField = styled((props: TextFieldProps) => (
+const CustomTextField = styled((props: TextFieldProps & Props) => (
   <TextFieldMui {...props} />
-))<Props>`
+)).attrs((props) => {
+  if (props.input) {
+    const { name, value, ...inputRest } = props.input;
+    return {
+      input: props.input,
+      name,
+      value,
+      inputProps: inputRest,
+      onChange: undefined,
+    };
+  } else {
+    return {
+      value: props.value,
+      onChange: props.onChange,
+    };
```

Comment:
As a quick feedback, it seems like you suggest to support two sets of incompatible props to the same element.
Would it maybe make sense to expose two different components then?

Reply:
It's a pretty common pattern in TypeScript 🤷‍♂️ It's like you do `string | number`, but a little bit more complex